core/src/main/java/io/grpc/Metadata.java

```diff
@@ -443,7 +443,7 @@ public void merge(Metadata other) {
     }
     int remaining = cap() - len();
     if (isEmpty() || remaining < other.len()) {
```

Comment:
It is needed since the underlying array could be null.  Merging two empty MDs would throw.